handler/handler.go

```diff
@@ -0,0 +1,33 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/lsolniczek/api-mock/models"
+)
+
+type HandlerFunc func(w http.ResponseWriter, r *http.Request)
+
+type handler struct {
+	HandlerFunc HandlerFunc
```

Comment:
Why would you want to pass `HandlerFunc` to the `handler`? If you made the HTTP-handling function as a method on the `*handler` and called it `ServeHTTP`, you'd implement an `http.Handler` interface:

```
type Handler interface {
        ServeHTTP(ResponseWriter, *Request)
}
```

So instead of returning `(*handler, error)`, you could return `(http.Handler, error)` and then pass it to `http.Handle` function in the caller (rather than `http.HandleFunc`.

Unless I'm missing an important part of the design here?